proposals/0000-swiftpm-binary-dependencies.md

```diff
@@ -0,0 +1,435 @@
+        enum Vendor: String, Encodable {
+            case apple
+            case ibm
+            case bgp
+            case suse
+            // And the rest
+        }
+
+        // Should be only the subset that Swift supports
+        enum OSType: String, Encodable {
+            case linux
+            case openBSD
+            case win32
+            case darwin
+            case iOS
+            case macOSX
+            // And the rest
+        }
+
+        let archType: ArchType
+        let vendor: Vendor
+        let osType: OSType
+        // Do we need the LLVM environment here?
+        public init(archType: ArchType, vendor: Vendor, osType: OSType) {
+            self.archType = archType
+            self.vendor = vendor
+            self.osType = osType
+        }
+    }
+
+    private let llvmTriplets: [LLVMTriplet]
+
+    private init(llvmTriplets [LLVMTriplet]) {
+        self.llvmTriplets = llvmTriplets
+    }
+
+    /// Create an artifact condition.
+    ///
+    /// - Parameters:
+    ///   - llvmTriplets: The llvm triplets for which this condition will be applied.
+    public static func when(
+        llvmTriplets: [LLVMTriplet]
+        ) -> ArtifactCondition {
+        return ArtifactCondition(llvmTriplets: llvmTriplets)
+    }
+}
+```
+
+### PackageDescription
+To opt out of binary packages we propose a new configuration point inside the package description.
+
+```swift
+public final class Package {
+    // ...
+    /// This disallows any binary dependency or any transitive binary dependency.
+    public var disallowsBinaryDependencies: Bool
+    // ...
+}
+```
+
+## New `Package.resolved` Behavior
+
+TODO: This still needs to be investigated.
+
+### Resolution
+
+Package resolution and dependency expression will not be impacted by this change (except where explicitly noted).
+
+## Binary Target Artifact Format
+
+SwiftPM supports various platforms and for each of them we need to find a format for the artifacts. Below is a list with a convention for the artifacts that we expect for each platform. (Courtesy to [Jake Petroules](https://forums.swift.org/t/pitch-support-for-binary-dependencies/27620/74?u=franzbusch))
+
+### Apple
+
+We use XCFrameworks. A single XCFramework can support any or all of the ABIs relevant for Apple OSes (macOS, iOS, tvOS, watchOS) as well as the special cases of Mac Catalyst and DriverKit.
+
+### Windows
+
+Nest artifacts inside architecture folders using the same naming convention as Microsoft does: ARM, ARM64, X86, X64.
+
+### Android
+
+Similar with Windows, nest inside architecture folders using the platform naming convention: armeabi-v7a, arm64-v8a, x86, x86_64.
+
+**Note:** That armeabi (armv5), mips and mips64 are obsolete since NDK r17; let's not worry about those.
+
+### Linux
+
+TODO: This section is still open for discussion since Linux brings a lot of its own problems with unstable ABI etc. with it.
+
+### Open discussion .o files
```

Comment:
We can remove this or move it to alternatives considered since we agreed on v1 only having XCFrameworks with the Xcode 11 feature set